docs/src/attributes.md

```diff
@@ -0,0 +1,23 @@
+# Attributes
+
+```@docs
+HDF5.Attribute
+attributes
+```
+
+## Mid-level Interface
+
+```@docs
+open_attribute
+create_attribute
+read_attribute
+write_attribute
+delete_attribute
+```
+
+## Convenience interface
+
+```@docs
+h5readattr
+h5writeattr
+```
```

Comment:
I don't know what to call these functions (honestly, these might be another good candidate for deprecation)